src/PSRule.Rules.Azure/rules/Azure.Template.Rule.ps1

```diff
@@ -304,6 +304,19 @@ Rule 'Azure.Template.ValidSecretRef' -Type '.json' -If { (IsParameterFile) } -Ta
     }
 }
 
+# Synopsis: Use comments for each resource to communicate purpose.
+Rule 'Azure.Template.UseComments' -Type '.json' -If { (IsTemplateFile) } -Tag @{ release = 'GA'; ruleSet = '2021_12'; } {
+    $resources = @(GetTemplateResources);
+
+    if ($resources.Length -eq 0) {
+        return $Assert.Pass();
+    }
+
+    foreach ($resource in $resources) {
+        $Assert.HasField($resource, 'comments');
+    }
```

Comment:
This is where a LINQ `All()` equivalent for PSRule would be usefule.

Reply:
@BernieWhite Nice. Some reason `AllOf` wasn't coming up in my intellisense at the time, seems to show up now.